replication-api-impl/src/main/java/com/connexta/replication/api/impl/persistence/pojo/ReplicationItemPojo.java

```diff
@@ -0,0 +1,404 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.api.impl.persistence.pojo;
+
+import com.connexta.ion.replication.api.Action;
+import com.connexta.ion.replication.api.Status;
+import java.util.Date;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.springframework.data.solr.core.mapping.Indexed;
+import org.springframework.data.solr.core.mapping.SolrDocument;
+
+/**
+ * This class provides a pojo implementation for a replication item capable of reloading all
+ * supported fields for all supported versions from the database. It also provides the capability of
+ * persisting back the fields based on the latest version format.
+ */
+@SolrDocument(collection = "replication_item")
+public class ReplicationItemPojo extends Pojo<ReplicationItemPojo> {
+  /**
+   * List of possible versions:
+   *
+   * <ul>
+   *   <li>1 - initial version.
+   * </ul>
+   */
+  public static final int CURRENT_VERSION = 1;
+
+  @Indexed(name = "metadata_id", type = "string")
```

Comment:
Wasn't the type already defined in the original class? I simply copied that over. As for being unsearchable, I guess simply because I didn't think we would ever need to search based on them. I can definitely revert that if you think we would need to search for those particular fields.

Reply:
Does the query by configId and status in the itemRepo class still work if status isn't searchable? Other than that I can't think of a specific reason we would search for another field so I think it's fine to leave them as is.

Reply:
Nice catch. This is exactly why I am working on integration test cases :-)